### PR TITLE
Updating documentation to display an import_format functionality + broken docs link

### PR DESCRIPTION
**Pull request by NickElliot**
Currently the % token functionality is only clarified in terraform.rb, which is not a place most users check. `resource.rb` is linked in our docs and will make this much more visible to users, while also suggesting the usage case of parent fields to avoid the misunderstanding that custom_imports are as necessary (https://github.com/hashicorp/terraform-provider-google/issues/12060).

If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [ ] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: mmv1/api/resource.rb [hunk 1]
```diff
@@ -147,6 +147,10 @@ module Properties
       # The Terraform resource id format used when calling #setId(...).
       # For instance, `{{name}}` means the id will be the resource name.
       attr_reader :id_format
+      # Sets the formats used to generate regex strings that match templated
```
Comment (rileykarson) [inline mmv1/api/resource.rb:150]: `import_format` is an override, the id / self_link generally create decent ones. We should make it clear this is used to handle corner cases, not in most cases.

**Review by rileykarson — APPROVED** [review]
One small nit, otherwise lgtm

File: mmv1/api/resource.rb (hunk 1 above)
Comment (NickElliot) [reply mmv1/api/resource.rb]: Updated the phrasing, let me know if this sounds good

Comment (rileykarson) [reply mmv1/api/resource.rb]: SGTM!

**Pull request merged**